pkg/chunk/purger/purger.go

```diff
@@ -151,6 +150,24 @@ func (dp *DataPurger) init(ctx context.Context) error {
 	return dp.loadInprocessDeleteRequests()
 }
 
+func (dp *DataPurger) loop(ctx context.Context) error {
+	loadRequestsTicker := time.NewTicker(time.Hour)
+	defer loadRequestsTicker.Stop()
+
+	for {
+		select {
+		case <-loadRequestsTicker.C:
```

Comment:
Nothing currently happens on tick, which I assume is not what you want. Unfortunately, [`fallthrough` or multi-select is not an option](https://github.com/golang/go/issues/23196) :-(

Reply:
hmm, sorry my bad. I thought it would work the same as fallthrough in switch statement. Will fix it.

Reply:
Unit test my be a good idea here :)